src/Pages/Shoppingbag/Shoppingbag.js

```diff
@@ -0,0 +1,239 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import Nav from "../../Components/Nav";
+import Footer from "../../Components/Footer";
+import "../../Styles/common.scss";
+import "./Shoppingbag.scss";
+import "../../Styles/reset.scss";
+import "./ItemList";
+import ItemList from "./ItemList";
+
+export default class Shoppingbag extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      countNumber: {},
+      pickItem: [...pickedItem].map((el) => ({ ...el, active: false })),
+      totalPrice: 0,
+      realtotalPrice: null,
+      shippingFee: 2500,
+      checkItem: false,
+      emptyDisplay: false,
+    };
+  }
+
+  handleCounting = (item, inDecrement, e) => {
+    e.preventDefault();
+    const minusValue = this.state.pickItem.map((el) =>
+      el.name === item.name
+        ? {
+            ...el,
+            price: parseInt((el.price * (el.count - 1)) / el.count),
+            count: el.count + inDecrement,
+          }
+        : el
+    );
+
+    const plusValue = this.state.pickItem.map((el) =>
+      el.name === item.name
+        ? {
+            ...el,
+            price: (el.price * (el.count + 1)) / el.count,
+            count: el.count + inDecrement,
+          }
+        : el
+    );
+
+    if (e.target.name === "plusButton" && item.count === 5) {
+      alert("최대주문수량 5개");
+      return;
+    }
+    if (e.target.name === "plusButton") {
+      this.setState(
+        {
+          pickItem: plusValue,
+        },
+        () => this.calculatePrice()
+      );
+    }
+    if (e.target.name === "minusButton" && item.count === 1) {
+      return;
+    }
+    if (e.target.name === "minusButton") {
+      this.setState(
+        {
+          pickItem: minusValue,
+        },
+        () => this.calculatePrice()
+      );
+    }
+  };
+
+  checkItem = (item) => {
+    this.setState({
+      pickItem: this.state.pickItem.map((el) =>
+        el.name === item.name ? { ...el, active: true } : el
+      ),
+    });
+  };
```

Comment:
```suggestion
  checkItem = (item) => {
    const newPickItem = this.state.pickItem.map(el => el.name === item.name ? { ...el, active: true } : el)
    this.setState({ pickItem: newPickItem });
  };
```

이런식으로 변수에 담아서 setState 함수의 값으로 지정해주면 훨씬 가독성이 좋아집니다.

Reply:
넵 :)